src/vecs/collection.py

```diff
@@ -340,33 +340,46 @@ def fetch(self, ids: Iterable[str]) -> List[Record]:
                     records.extend(chunk_records)
         return records
 
-    def delete(self, ids: Iterable[str]) -> List[str]:
+    def delete(self, ids: Optional[Iterable[str]] = None, filters: Optional[Metadata] = None) -> List[str]:
         """
-        Deletes vectors from the collection by their identifiers.
+        Deletes vectors from the collection by matching filters or ids.
 
         Args:
             ids (Iterable[str]): An iterable of vector identifiers.
+            filters (Metadata): A dictionary of metadata key-value pairs to match.
 
         Returns:
             List[str]: A list of the identifiers of the deleted vectors.
         """
-        if isinstance(ids, str):
-            raise ArgError("ids must be a list of strings")
+        if ids is None and filters is None:
+            raise VectorDeletionError("Either ids or metadata must be provided.")
```

Comment:
`VectorDeletionError` is not defined. Did you intended to add this in `exc.py`?
if so, it should inherit from ArgError to maintain compatibility